gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -111,13 +112,8 @@ void onApplyWindowInsets(WindowInsetsCompat insets) {
   }
 
   @Cold
-  private Observable<SignInState> saveUserOnSignIn(SignInState signInState) {
-    Optional<User> user = signInState.getUser();
-    if (user.isPresent()) {
-      return userRepository.saveUser(user.get()).andThen(Observable.just(signInState));
-    } else {
-      return Observable.just(signInState);
-    }
+  private Completable saveUser(SignInState signInState) {
+    return userRepository.saveUser(signInState.getUser().get());
```

Comment:
`.get()` -> `.orElseThrow()` ?

Reply:
Good point.. we shouldn't use `get()` unless we explicitly check whether the Optional is empty. The problem here is that the check would need to be done in the caller, not here together with the call to `get()`.

What if we consolidate this logic into our custom operators? Then the check and accessor are called in the same place, and the code is a bit more readable. Added a commit to demonstrate. Wdyt?